### DISCUSSION TITLE
How to downgrade python version back to 3.9

### DISCUSSION THREAD

**Discussion by macdrorepo** [Homebrew/discussions#3776] [Everyday usage] [answered]
Hello,

How I can downgrade python version on my macos from 3.10 to 3.9? 3.10 breaks some stuff internal to my company and of course I forget to suppress auto update. Brew list gets me:

`python@3.10
python@3.9`

Was trying to un-link 3.10 and link back 3.9 without success. Uninstall 3.10 can potentially break other packages

`Error: Refusing to uninstall /usr/local/Cellar/python@3.10/3.10.8
because it is required by ansible, ansible-lint, jsonschema, ldns, openssh, pipx, pre-commit and yamllint, which are currently installed.`

So how can I go back to 3.9?

**Comment by SMillerDev** [comment]
You can't downgrade packages, especially ones that other software depends on. The best way is just to put python 3.9 first in your path (or have your corporate software use `#!/usr/bin/env python@3.9` as shebang). To put python 3.9 first in your path you can do `export PATH="$(brew --prefix python@3.10)/bin:$PATH"`.

**Reply by macdrorepo** [reply]
So putting this to `export` to `zshrc` should solve the issues i guess?

**Reply by SMillerDev** [reply]
I'd start by running it once, but generally yes

**Reply by macdrorepo** [reply]
Thx didn't do a trick. So I've removed homebrew and want to get back to "original" python in MacOS. Now I'm getting:
`bad interpreter: /usr/local/opt/python@3.10/bin/python3.10: no such file or directory` of course as I've removed it. Do you know how to correct that maybe?

**Reply by SMillerDev** [reply]
Don't call scripts that depend on the Homebrew version of python.

**Reply by macdrorepo** [reply]
But I've removed whole hombrew seems some setup persists...

**Comment by macdrorepo** [comment, marked as answer]
So I did get rid of that by installing python from installer (3.9) and modify the PATH accordingly. 
Solved myslef.

**Reply by ademdc** [reply]
hey, could you specify what commands did you use?

**Comment by cowiekw** [comment]
I believe this works: 

`brew uninstall --ignore-dependencies python@3.10`
` brew install python@3.9`

just be mindful of dependencies that might break

**Comment by fedexin40** [comment]
My suggestion, use [pyenv](https://github.com/pyenv/pyenv) it was created for that

**Reply by gurvirbaraich** [reply]
@fedexin40  pyenv does not work for me. It stucks at xcode part

**Comment by HackerSSG** [comment]
if any one can not understand above method here is also method which i found and really also helpful you can check here : [https://codelikechamp.com/how-to-downgrade-python-3-11-to-3-10/](https://codelikechamp.com/how-to-downgrade-python-3-11-to-3-10/)

**Reply by osalbahr** [reply]
FYI, you may also use https://formulae.brew.sh/formula/python@3.9.

**Reply by HackerSSG** [reply]
oh nice

**Comment by HackerSSG** [comment]
hey could you share my link of this post downgrade python form 3.11 to 3.10 if any one can not understand above method here is also method which i found and really also helpful you can check here : https://codelikechamp.com/install-python-3-11-ubuntu/
      
        
        
          On Aug 28 2023, at 10:53 pm, Osama Albahrani ***@***.***> wrote:
          
        
        
          
FYI, you may also use ***@***.***

**Comment by HackerSSG** [comment]
sorry this one https://codelikechamp.com/how-to-downgrade-python-3-11-to-3-10/
      
        
        
          On Aug 28 2023, at 10:54 pm, Muhammad Usman ***@***.***> wrote:
          
        
        
          hey could you share my link of this post downgrade python form 3.11 to 3.10 if any one can not understand above method here is also method which i found and really also helpful you can check here : https://codelikechamp.com/install-python-3-11-ubuntu/
      
        
        
          On Aug 28 2023, at 10:53 pm, Osama Albahrani ***@***.***> wrote:
          
        
        
          
FYI, you may also use ***@***.***
